subworkflows/local/reference_indexing_multi.nf

```diff
@@ -2,101 +2,202 @@
 // Index input reference as required
 //
 
-include { GUNZIP                          } from '../../modules/nf-core/gunzip/main'
-include { BWA_INDEX                       } from '../../modules/nf-core/bwa/index/main'
-include { BOWTIE2_BUILD                   } from '../../modules/nf-core/bowtie2/build/main'
-include { SAMTOOLS_FAIDX                  } from '../../modules/nf-core/samtools/faidx/main'
-include { PICARD_CREATESEQUENCEDICTIONARY } from '../../modules/nf-core/picard/createsequencedictionary/main'
+include { GUNZIP as GUNZIP_FASTA          } from "../../modules/nf-core/gunzip/main"
+include { BWA_INDEX                       } from "../../modules/nf-core/bwa/index/main"
+include { BOWTIE2_BUILD                   } from "../../modules/nf-core/bowtie2/build/main"
+include { SAMTOOLS_FAIDX                  } from "../../modules/nf-core/samtools/faidx/main"
+include { PICARD_CREATESEQUENCEDICTIONARY } from "../../modules/nf-core/picard/createsequencedictionary/main"
 // TODO missing: circulargeneraotr?
 
-workflow REFERENCE_INDEXING_SINGLE {
+workflow REFERENCE_INDEXING_MULTI {
 
     take:
-    samplesheet // file: /path/to/name.{fasta,fa,fna,fas,fasta.gz,fa.gz,fna.gz,fas.gz}
+    referencesheet // file: /path/to/name.{csv,tsv}
 
     main:
     ch_versions = Channel.empty()
-    // TODO FINISH!
 
-    if ( samplesheet.extension == 'tsv' ){
-        ch_splitsamplesheet_for_branch = samplesheet
-                                            .csv
-                                            .splitCsv ( header:true, sep:'\t' )
+    // Parse CSV and detect files to load
```

Comment:
Should this be split into a conditional that reads the file into a channel, and then that channel goes into a single `.map`? Will be annoying to tweak every column 2x for each change we make to the reference tsv.